packages/web/resolvers/topic.ts

```diff
@@ -119,4 +128,68 @@ const TopicQueries = extendType({
   },
 })
 
-export default [TopicTranslation, Topic, TopicQueries]
+const TopicMutations = extendType({
+  type: 'Mutation',
+  definition(t) {
+    t.field('addUserInterest', {
+      type: 'UserInterest',
+      args: {
+        topicId: intArg({ required: true })
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+
+        if (!userId) {
+          throw new Error('You must be logged in to add a user interest.')
+        }
+
+        const topic = await ctx.db.topic.findUnique({
+          where: { id: args.topicId },
+        })
+
+        if (!topic) {
+          throw new Error(`Unable to find language with id "${args.topicId}".`)
+        }
+
+        return ctx.db.userInterest.create({
+          data: {
+            user: { connect: { id: userId } },
+            topic: { connect: { id: args.topicId } },
+          }
+        })
+      },
+    })
+    t.field('removeUserInterest', {
+      type: 'UserInterest',
+      args: {
+        topicId: intArg({ required: true })
+      },
+      resolve: async (_parent, args, ctx) => {
+        const { userId } = ctx.request
+
+        if (!userId) {
+          throw new Error('You must be logged in to add a user interest.')
+        }
+
+        const interestFilter = {
+          where: {
+            userId_topicId: {
+              topicId: args.topicId,
+              userId,
+            },
+          },
+        }
+
+        const userInterest = await ctx.db.userInterest.findUnique(interestFilter)
```

Comment:
Is checking for the existence of the user/topic relation. necessary? We fail if it doesn't exist, why not just attempt the delete and fail if the delete doesn't go through? That way we could save a round trip to the DB

Reply:
hmm, this is also done here: https://github.com/Journaly/journaly/blob/master/packages/web/resolvers/language.ts#L139

let me know if you would still advise to remove this check!

Reply:
Yeah, in both cases I think we should just do the delete. `db.delete()` will throw if no record is found for deletion, so we can either just let that exception bubble up or catch and rethrow our own exception, it's the same result either way.